hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -1051,4 +1030,55 @@ private void addCacheRegionDefinition(CacheRegionDefinition cacheRegionDefinitio
 			this.cacheRegionDefinitions.add( cacheRegionDefinition );
 		}
 	}
+
+	private <T> T loadSettingInstance(String settingName, Object settingValue, Class<T> clazz) {
+		if ( settingValue == null ) {
+			return null;
+		}
+
+		T instance = null;
+		Class<? extends T> instanceClass = null;
+
+		if ( clazz.isAssignableFrom( settingValue.getClass() ) ) {
```

Comment:
not sure if this should be `if ( clazz.isInstance( settingValue ) )` instead of `if ( clazz.isAssignableFrom( settingValue.getClass() ) )`